ext/ddtrace_profiling_native_extension/collectors_cpu_and_wall_time.c

```diff
@@ -759,3 +819,38 @@ static VALUE _native_stats(DDTRACE_UNUSED VALUE _self, VALUE collector_instance)
 
   return stats_as_ruby_hash(state);
 }
+
+static void trace_identifiers_for(struct cpu_and_wall_time_collector_state *state, VALUE thread, struct trace_identifiers *trace_identifiers_result) {
```

Comment:
Do we have the GVL when calling this function?

Reply:
Yes! Added additional documentation in c7f6401bdb066fa37c188509cf000c7ebbb90be2, but currently the whole sampling approach is done while holding the GVL.